src/hotspot/share/gc/g1/g1VMOperations.cpp

```diff
@@ -122,7 +122,7 @@ VM_G1CollectForAllocation::VM_G1CollectForAllocation(size_t         word_size,
 void VM_G1CollectForAllocation::doit() {
   G1CollectedHeap* g1h = G1CollectedHeap::heap();
 
-  if (_word_size > 0) {
+  if (_word_size > 0 && _gc_cause != GCCause::_g1_proactive_collection) {
```

Comment:
What do you think about adding a helper to ´VM_G1CollectForAllocation`:
```
bool try_allocate_before_gc() {
  if (_gc_cause == GCCause::_g1_proactive_collection) {
    // Never allocate before proactive GCs.
    return false;
  }
  return true;
}
```
The condition would then read a bit better:
```suggestion
  if (try_allocate_before_gc() && _word_size > 0) {
```